gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -227,7 +236,7 @@ def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
     if current_declaration.nil?
       return
     end
-    T::Private::DeclState.current.reset!
+    T::Private::DeclState.current.reset! unless attr_accessor_reader?(method_name)
```

Comment:
I chose to just leave the same sig for the writer method, but I guess I can alternatively, clear the sig unconditionally and add a duped and modified version of the `current_declaration` as `current` if this is an `attr_accessor` reader.

That would actually simplify the processing we need to do later in `Signature#initialize`. Happy to make the change if you think it makes more sense.

Reply:
Yeah, I think I would prefer that we create that signature here. It's a little more explicit and means that there's only one heuristic in play (the `attr_accessor` in the call stack) and not also "the signature is technically ill-formed but looks like a writer method".

Reply:
Ok made the change, but since we don't want to run the decl block here, i've instead chosen to create a cloned decl block whose `blk` is a proc that calls the `blk` of the current decl and then adds the params as it should be for the writer.

I am not sure if this is the best or most efficient way of doing this. Please let me know if it is not.